### ISSUE TITLE
Several plots throwing column length error (0.9.1)

### ISSUE THREAD

**Issue by mistakevin** [bokeh/bokeh#2491]
Horizon plots are throwing an `E-1000 (COLUMN_LENGTHS): ColumnDataSource column lengths are not all the same` error. Error not present in 0.9.0. Graph still renders.

**Comment by bryevdv** [comment]
@mistakevin there are lots of places where errors are triggered in examples (one reason we are only reporting errors for now, not raising exceptions on them). Let's make this an issue to root out all of them

**Comment by damianavila** [comment]
This is probably the new validation making its work... we don't have any hard-constrains on the column lengths, we assume that as fact, but we don't do anything to force it... and now this is detected by the validation machinery...

**Comment by damianavila** [comment]
>Let's make this an issue to root out all of them

I agree...

**Comment by aavanian** [comment]
> we don't have any hard-constrains on the column lengths

Maybe I read this wrong but actually, even if it's not a hard constraint to have same column length in a `ColumnDataSource`, it definitely has an impact. If the same source is used on different plots using columns of different sizes, it seems the extra lines are being cut-off:

```python
        p1 = figure(x_axis_type="datetime", x_range=self._x_range, y_range=self._y1_range)
        p1.line(x='date', y='value', source=self.source)

        p2 = figure(x_axis_type="datetime", x_range=p1.x_range, y_range=self._y2_range)
        p2.patch(x='date_patch', y='value2_patch', source=self.source)
```

where `date`, `value`, `value2` are the original data (all the same length) and `date_patch` and `value2_patch` are derived from `date` and `value2` to close the patch correctly, i.e. adding two elements representing coordinates (`source.data['date'][-1]`, `0`) and (`source.data['date'][0]`, `0`).

With that, the patch is actually closed by joining (`source.data['date'][-1]`, `source.data['value2'][-1]`) and (`source.data['date'][0]`, `source.data['value2'][0]`) which I assume happens because it discards the extra items in `source.data['date_patch']` and `source.data['spread_patch']`.

If I create another `ColumnDataSource` with the derived columns only and point `p2.patch()` to it, then it works as expected.

**Comment by bryevdv** [comment]
> Maybe I read this wrong but actually, even if it's not a hard constraint to have same column length in a ColumnDataSource, it definitely has an impact. 

Yes it would be better to say "enforcement", perhaps. It is absolutely a *requirement* and an *mandatory assumption* of `ColumnDataSource` objects that all the columns have the same length. It is an error to have ragged column lengths. Right now we display the validation error for this at serialization time (it's not possible to detect earlier, in principal), but we have not yet turned on raising an exception. At that point I supposed you can say there will be "enforcement".

**Comment by bryevdv** [comment]
Burtin still does this, at a minimum.

**Comment by canavandl** [comment]
Appears to be resolved by PR #3406 . Master doesn't raise this warn while building the Sphinx docs.
